src/Renci.SshNet/ConnectionInfo.cs

```diff
@@ -124,24 +124,9 @@ public class ConnectionInfo : IConnectionInfoInternal
         public ProxyTypes ProxyType { get; private set; }
 
         /// <summary>
-        /// Gets proxy connection host.
+        /// Gets connection info used for connecting through a proxy.
         /// </summary>
-        public string ProxyHost { get; private set; }
-
-        /// <summary>
-        /// Gets proxy connection port.
-        /// </summary>
-        public int ProxyPort { get; private set; }
-
-        /// <summary>
-        /// Gets proxy connection username.
-        /// </summary>
-        public string ProxyUsername { get; private set; }
-
-        /// <summary>
-        /// Gets proxy connection password.
-        /// </summary>
-        public string ProxyPassword { get; private set; }
```

Comment:
These should all stay (no need to break anyone here), they can forward to the other ConnectionInfo if it exists

Reply:
I can return these if you want. I removed them to prevent confusion. Now, both SSH and HTTP/SOCKS connections may now be used as a proxy connection. However, SSH connection authentication is handled differently from HTTP/SOCKS proxies, with the latter keeping the username and password in the ConnectionInfo instance, but SSH inside one or more child `AuthenticationMethod`. Having the ProxyUsername and ProxyPassword may suggest these would also represent the respective fields inside a password authentication for SSH, but this is then not the case.